pkg/bpf/map.go

```diff
@@ -512,3 +514,23 @@ func ConvertKeyValue(bKey []byte, bValue []byte, key interface{}, value interfac
 
 	return nil
 }
+
+// MetadataDiff compares the metadata of the BPF maps and returns false if the
+// metadata does not match
+func (m *Map) MetadataDiff(other *Map) bool {
+	if m == nil || other == nil {
```

Comment:
Oh, would both being nil be a `true`?

Reply:
Thought about it, doesn't really make sense.